src/shared/HOC/body-disable-HOC.js

```diff
@@ -0,0 +1,21 @@
+import React, { Component } from "react";
+
+const BODY_DISABLED_CLASS = "body--disabled";
+
+function bodyDisableHOC(WrappedComponent) {
+  return class extends Component {
+    componentDidMount() {
+      document.body.classList.add(BODY_DISABLED_CLASS);
```

Comment:
Interesting idea. But this does not work on the iPhone. And the menu opens with a very long delay.